### PR TITLE
Fix for network failures in WorksForYou

### LINKED ISSUE #1425: Reload WorksForYou upon network request failure

### PR DESCRIPTION
**Pull request by sarahscott**
If the network connection is lost, WorksForYouVC will reload if the user navigates elsewhere and returns with a solid connection.

Fixes #1425

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>PR is classed as Work in Progress</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by alloy** [comment]
Can you add tests for both VCs and a CHANGELOG?

**Comment by sarahscott** [comment]
On it!

File: Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m [hunk 1]
```diff
@@ -60,6 +61,19 @@ @interface ARWorksForYouViewController ()
...
+
+describe(@"handling network failures", ^{
+    before(^{
+        [OHHTTPStubs stubJSONResponseAtPath:@"/api/v1/me/notifications" withResponse:[OHHTTPStubsResponse responseWithError:[NSError errorWithDomain:NSURLErrorDomain code:404 userInfo:nil]]];
+    });
+    
+    it(@"sets networkingDidFail to YES", ^{
+        subject.worksForYouNetworkModel = [[ARWorksForYouNetworkModel alloc] init];
+        [subject beginAppearanceTransition:YES animated:NO];
+        expect(subject.networkingDidFail).to.beTruthy();
```
Comment (sarahscott) [inline Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m]: Any idea why the failure block wouldn't be called?

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m [hunk 2]
```diff
@@ -166,7 +169,11 @@ - (void)getNextItemSet
         } else if (sself.shouldShowEmptyState) {
             [sself showEmptyState];
         }
-    } failure:nil];
+    } failure:^(NSError *error) {
+        if (self.worksForYouNetworkModel.currentPage == 1) {
+            self.networkingDidFail = YES;
```
Comment (sarahscott) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: I set a breakpoint here and the test never actually runs this block. I've checked to make sure the OHTTPStub is working.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>PR is classed as Work in Progress</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m [hunk 3]
```diff
@@ -60,6 +61,24 @@ @interface ARWorksForYouViewController ()
...
+    });
+});
+
+describe(@"handling network failures", ^{
+    it(@"sets networkingDidFail to YES", ^{
+        subject.worksForYouNetworkModel = stubbedNetworkModel;
+        
+        id stub = [OCMockObject partialMockForObject:subject.worksForYouNetworkModel];
+        
+        [[[stub stub] andDo:^(NSInvocation *invocation) {
```
Comment (sarahscott) [inline Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m]: I'll change the name to something that's not stub...

File: Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m (hunk 1 above)
Comment (orta) [inline Artsy_Tests/View_Controller_Tests/ARWorksForYouViewControllerTests.m]: Double check that this is happening as you expect here: https://github.com/artsy/eigen/blob/master/Artsy_Tests/Stubs/ArtsyOHHTTPAPI.m#L105-L107

**Comment by orta** [comment]
This feels a bit off, it might make more sense to move the `+@property (nonatomic, assign) BOOL networkingDidFail;` to the `networkModel`, as it's more the domain of that then the view controller. This also means you can get rid of those OCMocks in the tests too, which is a 👍

**Comment by alloy** [comment]
@orta Good point.

**Comment by sarahscott** [comment]
👌

**Comment by sarahscott** [comment]
🍏

**Comment by ashfurrow** [comment]
💯

**Comment by alloy** [comment]
Noice! 👌

**Pull request merged**